scripts/format-staged.ps1

```
@@ -0,0 +1,22 @@
+$staged = git diff --name-only --cached --diff-filter=ACM
+$rs = $staged | Where-Object { $_ -match '\.rs$' }
+if (-not $rs) { Write-Host "No staged Rust files to format."; exit 0 }
+
+$fmtFound = (Get-Command rustfmt -ErrorAction SilentlyContinue) -ne $null
+foreach ($f in $rs) {
+    if ($fmtFound) {
+        Write-Host "Formatting (rustfmt): $f" -ForegroundColor Cyan
+        rustfmt --edition 2021 $f
+    }
+    else {
+        Write-Host "Formatting (cargo fmt fallback)" -ForegroundColor Yellow
```

PSScriptAnalyzer (1.23.0)

[warning] File 'format-staged.ps1' uses Write-Host. Avoid using Write-Host because it might not work in all hosts, does not work when there is no host, and (prior to PS 5.0) cannot be suppressed, captured, or redirected. Instead, use Write-Output, Write-Verbose, or Write-Information.

(PSAvoidUsingWriteHost)